File: admin/src/api/wallet.ts
import { http, type ApiEnvelope } from '../utils/request'
import type { PaginatedData } from './admin-user'

export interface WalletUserSummary {
  id: number
  username: string
  email: string
  display_name: string | null
}

export interface WalletItem {
  wallet_no: string
  user: WalletUserSummary
  currency: string
  status: 'active' | 'disabled'
  available_balance_cents: number
  total_recharged_cents: number
  total_spent_cents: number
  total_refunded_cents: number
  created_at: string
  updated_at: string
}

export interface WalletLedgerItem {
  entry_no: string
  wallet_no: string
  user: WalletUserSummary
  direction: 'credit' | 'debit'
  entry_type: 'recharge' | 'payment' | 'refund'
  amount_cents: number
  balance_before_cents: number
  balance_after_cents: number
  currency: string
  related_type: string
  related_no: string
  summary: string | null
  created_at: string
}

export interface WalletRechargeItem {
  recharge_no: string
  wallet_no: string
  user: WalletUserSummary
  provider: 'alipay' | 'wechat'
  method: 'alipay_page' | 'alipay_wap' | 'wechat_native' | 'wechat_h5'
  status: 'pending' | 'paid' | 'closed' | 'failed'
  amount_cents: number
  currency: string
  expires_at: string
  paid_at: string | null
  closed_at: string | null
  failed_at: string | null
  created_at: string
}

export interface WalletDetail extends WalletItem {
  recent_ledger: WalletLedgerItem[]
  recent_recharges: WalletRechargeItem[]
}

export async function getWallets(params?: Record<string, unknown>) {
  const response = await http.get<ApiEnvelope<PaginatedData<WalletItem>>>('/wallets', { params })
  return response.data.data
}

export async function getWalletDetail(walletNo: string) {
  const response = await http.get<ApiEnvelope<WalletDetail>>(`/wallets/${walletNo}`)
  return response.data.data
}

export async function getWalletLedger(params?: Record<string, unknown>) {
  const response = await http.get<ApiEnvelope<PaginatedData<WalletLedgerItem>>>('/wallet-ledger', { params })
  return response.data.data
}

export async function getWalletRecharges(params?: Record<string, unknown>) {
  const response = await http.get<ApiEnvelope<PaginatedData<WalletRechargeItem>>>('/wallet-recharges', { params })
  return response.data.data
}

File: server/internal/usecase/admin/dto/wallet.go
package dto

import "time"

type WalletListQuery struct {
	Page        int    `form:"page" validate:"omitempty,min=1"`
	PerPage     int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	WalletNo    string `form:"wallet_no" validate:"omitempty,max=64"`
	Status      string `form:"status" validate:"omitempty,oneof=active disabled"`
	UserKeyword string `form:"user_keyword" validate:"omitempty,max=128"`
}

type WalletLedgerListQuery struct {
	Page        int    `form:"page" validate:"omitempty,min=1"`
	PerPage     int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	WalletNo    string `form:"wallet_no" validate:"omitempty,max=64"`
	UserKeyword string `form:"user_keyword" validate:"omitempty,max=128"`
	Direction   string `form:"direction" validate:"omitempty,oneof=credit debit"`
	EntryType   string `form:"entry_type" validate:"omitempty,oneof=recharge payment refund"`
	RelatedNo   string `form:"related_no" validate:"omitempty,max=64"`
	DateFrom    string `form:"date_from" validate:"omitempty,max=32"`
	DateTo      string `form:"date_to" validate:"omitempty,max=32"`
}

type WalletRechargeListQuery struct {
	Page        int    `form:"page" validate:"omitempty,min=1"`
	PerPage     int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	WalletNo    string `form:"wallet_no" validate:"omitempty,max=64"`
	UserKeyword string `form:"user_keyword" validate:"omitempty,max=128"`
	Provider    string `form:"provider" validate:"omitempty,oneof=alipay wechat"`
	Method      string `form:"method" validate:"omitempty,oneof=alipay_page alipay_wap wechat_native wechat_h5"`
	Status      string `form:"status" validate:"omitempty,oneof=pending paid closed failed"`
	RechargeNo  string `form:"recharge_no" validate:"omitempty,max=64"`
	DateFrom    string `form:"date_from" validate:"omitempty,max=32"`
	DateTo      string `form:"date_to" validate:"omitempty,max=32"`
}

type WalletUserSummary struct {
	ID          uint64  `json:"id"`
	Username    string  `json:"username"`
	Email       string  `json:"email"`
	DisplayName *string `json:"display_name"`
}

type WalletItem struct {
	WalletNo              string            `json:"wallet_no"`
	User                  WalletUserSummary `json:"user"`
	Currency              string            `json:"currency"`
	Status                string            `json:"status"`
	AvailableBalanceCents uint64            `json:"available_balance_cents"`
	TotalRechargedCents   uint64            `json:"total_recharged_cents"`
	TotalSpentCents       uint64            `json:"total_spent_cents"`
	TotalRefundedCents    uint64            `json:"total_refunded_cents"`
	CreatedAt             time.Time         `json:"created_at"`
	UpdatedAt             time.Time         `json:"updated_at"`
}

type WalletLedgerItem struct {
	EntryNo            string            `json:"entry_no"`
	WalletNo           string            `json:"wallet_no"`
	User               WalletUserSummary `json:"user"`
	Direction          string            `json:"direction"`
	EntryType          string            `json:"entry_type"`
	AmountCents        uint64            `json:"amount_cents"`
	BalanceBeforeCents uint64            `json:"balance_before_cents"`
	BalanceAfterCents  uint64            `json:"balance_after_cents"`
	Currency           string            `json:"currency"`
	RelatedType        string            `json:"related_type"`
	RelatedNo          string            `json:"related_no"`
	Summary            *string           `json:"summary"`
	CreatedAt          time.Time         `json:"created_at"`
}

type WalletRechargeItem struct {
	RechargeNo  string            `json:"recharge_no"`
	WalletNo    string            `json:"wallet_no"`
	User        WalletUserSummary `json:"user"`
	Provider    string            `json:"provider"`
	Method      string            `json:"method"`
	Status      string            `json:"status"`
	AmountCents uint64            `json:"amount_cents"`
	Currency    string            `json:"currency"`
	ExpiresAt   time.Time         `json:"expires_at"`
	PaidAt      *time.Time        `json:"paid_at"`
	ClosedAt    *time.Time        `json:"closed_at"`
	FailedAt    *time.Time        `json:"failed_at"`
	CreatedAt   time.Time         `json:"created_at"`
}

type WalletDetail struct {
	WalletItem
	RecentLedger    []WalletLedgerItem   `json:"recent_ledger"`
	RecentRecharges []WalletRechargeItem `json:"recent_recharges"`
}
